modules/tensor_mechanics/src/materials/ComputeVariableIsotropicElasticityTensor.C

```diff
@@ -0,0 +1,66 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeVariableIsotropicElasticityTensor.h"
+
+template<>
+InputParameters validParams<ComputeVariableIsotropicElasticityTensor>()
+{
+  InputParameters params = validParams<ComputeIsotropicElasticityTensor>();
+  params.addClassDescription("Compute an isotropic elasticity tensor for elastic constants that change as a function of temperature");
+  params.addRequiredCoupledVar("temperature", "Coupled temperature");
+  params.addParam<bool>("store_old_elasticity_tensor", true, "Parameter to save the old elasticity tensor values, set to true when the elasticity tensor components change, i.e. with temperature");
+  return params;
+}
+
+ComputeVariableIsotropicElasticityTensor::ComputeVariableIsotropicElasticityTensor(const InputParameters & parameters) :
+    ComputeIsotropicElasticityTensor(parameters),
+    _temperature(coupledValue("temperature")),
+    _elasticity_tensor_old(declarePropertyOld<RankFourTensor>(_elasticity_tensor_name))
+{
+  if (_lambda_set && _shear_modulus_set)
+  {
+    _youngs_modulus = _shear_modulus * (3.0 * _lambda + 2.0 * _shear_modulus) / (_lambda + _shear_modulus);
+    _poissons_ratio = _lambda / (2.0 * (_lambda + _shear_modulus));
+  }
+  else if (_shear_modulus_set && _bulk_modulus_set)
+  {
+    _youngs_modulus = 9.0 * _bulk_modulus * _shear_modulus / (3.0 * _bulk_modulus + _shear_modulus);
+    _poissons_ratio = (3.0 * _bulk_modulus - 2.0 * _shear_modulus) / (2.0 * (3.0 * _bulk_modulus + _shear_modulus));
+  }
+}
+
+void
+ComputeVariableIsotropicElasticityTensor::initQpStatefulProperties()
+{
+  _elasticity_tensor_old[_qp] = _elasticity_tensor[_qp];
```

Comment:
Yes the method in the Material BASE CLASS has a warning in it's implementation so you have no choice but to supply an empty method. What this means is that you are fine with having zeros in your older property when you first access it. If that's the case, you're good!